### PR TITLE
Improvements/ Removed unused interfaces/functions from classes

### PR DESCRIPTION
**Pull request by wasif1**
1. Removed from all classes public ngOnInit() {}
2. Removed ngOnInit() method from all files
3. Removed the OnInit interface from the classes
4. Removed the OnInit import from the top of all files

### CODE REVIEW

**Comment by dw-0** [comment]
Thanks. Please rebase your feature branch to the current develop branch and resolve any remaining conflicts :)

**Comment by wasif1** [comment]
All conflicts resolved please check now

File: apps/client/src/app/components/admin-jobs/admin-jobs.component.ts [hunk 1]
```diff
@@ -29,7 +28,7 @@ import { takeUntil } from 'rxjs/operators';
   styleUrls: ['./admin-jobs.scss'],
   templateUrl: './admin-jobs.html'
 })
-export class AdminJobsComponent implements OnDestroy, OnInit {
```
Comment (dw-0) [inline apps/client/src/app/components/admin-jobs/admin-jobs.component.ts:32]: OnInit is actually implemented in this file. Do not remove it here

File: apps/client/src/app/components/admin-market-data/admin-market-data.component.ts [hunk 2]
```diff
@@ -51,8 +50,7 @@ import { CreateAssetProfileDialogParams } from './create-asset-profile-dialog/in
   templateUrl: './admin-market-data.html'
 })
 export class AdminMarketDataComponent
-  implements AfterViewInit, OnDestroy, OnInit
```
Comment (dw-0) [inline apps/client/src/app/components/admin-market-data/admin-market-data.component.ts:54]: OnInit is actually implemented in this file. Do not remove it here

File: apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.component.ts [hunk 3]
```diff
@@ -44,7 +43,7 @@ import { AssetProfileDialogParams } from './interfaces/interfaces';
   templateUrl: 'asset-profile-dialog.html',
   styleUrls: ['./asset-profile-dialog.component.scss']
 })
-export class AssetProfileDialog implements OnDestroy, OnInit {
```
Comment (dw-0) [inline apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.component.ts:47]: OnInit is actually implemented in this file. Do not remove it here

**Review by dw-0 — CHANGES_REQUESTED** [review]
Hi, i reviewed a few files but spotted some issues. You are removing the `implements OnInit` from several files where ngOnInit is actually implemented AND the content of it is not empty. Removing `implements OnInit` from such files is not correct.

Please doublecheck all modified files and make sure the modifications are actually necessary and specified as per the task

**Comment by wasif1** [comment]
I pushed the updated code again, do i need to create another pull request or this same request is fine?

**Comment by dw-0** [comment]
The same PR is fine, it will update accordingly.

File: apps/client/src/app/components/dialog-footer/dialog-footer.component.ts [hunk 4]
```diff
@@ -14,12 +13,12 @@ import {
...
   styleUrls: ['./dialog-footer.component.scss']
 })
-export class DialogFooterComponent implements OnInit {
+export class DialogFooterComponent {
   @Input() deviceType: string;
 
   @Output() closeButtonClicked = new EventEmitter<void>();
 
-  public ngOnInit() {}
+  public constructor() {}
```
Comment (dw-0) [inline apps/client/src/app/components/dialog-footer/dialog-footer.component.ts:21]: Do not add an empty constructor

File: apps/client/src/app/components/dialog-header/dialog-header.component.ts [hunk 5]
```diff
@@ -14,14 +13,14 @@ import {
...
-export class DialogHeaderComponent implements OnInit {
+export class DialogHeaderComponent {
   @Input() deviceType: string;
   @Input() position: 'center' | 'left' = 'left';
   @Input() title: string;
 
   @Output() closeButtonClicked = new EventEmitter<void>();
 
-  public ngOnInit() {}
+  public constructor() {}
```
Comment (dw-0) [inline apps/client/src/app/components/dialog-header/dialog-header.component.ts:23]: Do not add an empty constructor

File: apps/client/src/app/components/fear-and-greed-index/fear-and-greed-index.component.ts [hunk 6]
```diff
@@ -15,13 +14,13 @@ import {
...
 })
-export class FearAndGreedIndexComponent implements OnChanges, OnInit {
+export class FearAndGreedIndexComponent implements OnChanges{
   @Input() fearAndGreedIndex: number;
 
   public fearAndGreedIndexEmoji: string;
   public fearAndGreedIndexText: string;
 
-  public ngOnInit() {}
+  public constructor() {}
```
Comment (dw-0) [inline apps/client/src/app/components/fear-and-greed-index/fear-and-greed-index.component.ts:23]: Do not add an empty constructor

File: apps/client/src/app/components/portfolio-summary/portfolio-summary.component.ts [hunk 7]
```diff
@@ -35,7 +34,7 @@ export class PortfolioSummaryComponent implements OnChanges, OnInit {
   );
   public timeInMarket: string;
 
-  public ngOnInit() {}
+  public constructor() {}
```
Comment (dw-0) [inline apps/client/src/app/components/portfolio-summary/portfolio-summary.component.ts:37]: Do not add an empty constructor

File: apps/client/src/app/components/toggle/toggle.component.ts [hunk 8]
```diff
@@ -34,7 +33,7 @@ export class ToggleComponent implements OnChanges, OnInit {
 
   public optionFormControl = new FormControl<string>(undefined);
 
-  public ngOnInit() {}
+  public constructor() {}
```
Comment (dw-0) [inline apps/client/src/app/components/toggle/toggle.component.ts:36]: Do not add an empty constructor

File: apps/client/src/app/pages/faq/self-hosting/self-hosting-page.component.ts [hunk 9]
```diff
@@ -10,7 +10,7 @@ import { Subject } from 'rxjs';
 export class SelfHostingPageComponent implements OnDestroy {
   private unsubscribeSubject = new Subject<void>();
 
-  public ngOnInit() {}
+  public constructor() {}
```
Comment (dw-0) [inline apps/client/src/app/pages/faq/self-hosting/self-hosting-page.component.ts:13]: Do not add an empty constructor

File: apps/client/src/app/pages/i18n/i18n-page.component.ts [hunk 10]
```diff
@@ -8,10 +8,10 @@ import { Subject } from 'rxjs';
   styleUrls: ['./i18n-page.scss'],
   templateUrl: './i18n-page.html'
 })
-export class GfI18nPageComponent implements OnInit {
+export class GfI18nPageComponent {
   private unsubscribeSubject = new Subject<void>();
 
-  public ngOnInit() {}
+  public constructor() {}
```
Comment (dw-0) [inline apps/client/src/app/pages/i18n/i18n-page.component.ts:14]: Do not add an empty constructor

File: apps/client/src/app/pages/markets/markets-page.component.ts [hunk 11]
```diff
@@ -7,10 +7,10 @@ import { Subject } from 'rxjs';
   styleUrls: ['./markets-page.scss'],
   templateUrl: './markets-page.html'
 })
-export class MarketsPageComponent implements OnDestroy, OnInit {
+export class MarketsPageComponent implements OnDestroy{
   private unsubscribeSubject = new Subject<void>();
 
-  public ngOnInit() {}
+  public constructor() {}
```
Comment (dw-0) [inline apps/client/src/app/pages/markets/markets-page.component.ts:13]: Do not add an empty constructor

**Review by dw-0 — CHANGES_REQUESTED** [review]
Please do not re-introduce empty constructors :) they just got removed in a previous PR

**Comment by wasif1** [comment]
I rebase the project first then pushed the code. i'll check empty constructor

**Comment by wasif1** [comment]
Removed the empty constructors aswell, please check

**Comment by dw-0** [comment]
The code formatting check might fail now. Please run this on the project: https://github.com/ghostfolio/ghostfolio/blob/main/package.json#L34

Then @dtslvr should have a look at it

**Comment by wasif1** [comment]
Screenshot 2024-10-19 at 10 13 19 PM

I tried to run this command, even after updating npm. still not working :/

**Review by dtslvr — APPROVED** [review]
Thanks for the contribution, @wasif1.
I have fixed the formatting. Ready to merge now.

**Comment by wasif1** [comment]
Thankyou so much @dtslvr and @dw-0  🏆
Im new in this open source world and this is my first PR. 👍 
Excited to do more contributions. 🚀

**Comment by dtslvr** [comment]
> Can you add hacktoberfest label aswell?

The issue https://github.com/ghostfolio/ghostfolio/issues/3959 where you are assigned is already labeled with `hacktoberfest`. Please get in touch again if your contribution does not appear within a day in your _Hacktoberfest_ account.

**Pull request merged**